### PR TITLE
Extract shared Lottie playback from MAUI

### PR DESCRIPTION
**Pull request by mattleibow**
## Summary

Extracts the reusable Lottie playback layer from the MAUI control into `SkiaSharp.Extended`.

- Adds the non-owning `SKLottiePlayer`, `SKLottieRepeat`, passive `SKLottieAnimation` result, and shared internal loader to the core package.
- Refactors `SKLottieView` and its file, URI, and stream sources to delegate playback/loading while retaining MAUI binding, source-change, handler lifecycle, cancellation, disposal, and public events.
- Moves focused repeat/playback coverage to core, retains MAUI adapter tests, and documents the core player and MAUI integration.

This bottom layer intentionally excludes Blazor, samples, DevFlow tooling, and the frame-counter move.

## Validation

- `dotnet build source/SkiaSharp.Extended.UI.Maui/SkiaSharp.Extended.UI.Maui.csproj --configuration Release --framework net10.0`
- `dotnet test tests/SkiaSharp.Extended.Tests/SkiaSharp.Extended.Tests.csproj --configuration Release --filter 'FullyQualifiedName~SKLottie'`
- `dotnet test tests/SkiaSharp.Extended.UI.Maui.Tests/SkiaSharp.Extended.UI.Maui.Tests.csproj --configuration Release --filter 'FullyQualifiedName~Lottie'`
- `cd docs && docfx build` (succeeds with existing DocFX warnings)

### CODE REVIEW

File: source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1 +1,37 @@
...
+SkiaSharp.Extended.SKLottiePlayer.Animation.set -> void
+SkiaSharp.Extended.SKLottiePlayer.AnimationCompleted -> System.EventHandler?
+SkiaSharp.Extended.SKLottiePlayer.AnimationSpeed.get -> double
+SkiaSharp.Extended.SKLottiePlayer.AnimationSpeed.set -> void
+SkiaSharp.Extended.SKLottiePlayer.AnimationUpdated -> System.EventHandler?
+SkiaSharp.Extended.SKLottiePlayer.Duration.get -> System.TimeSpan
+SkiaSharp.Extended.SKLottiePlayer.HasAnimation.get -> bool
+SkiaSharp.Extended.SKLottiePlayer.IsComplete.get -> bool
+SkiaSharp.Extended.SKLottiePlayer.Progress.get -> System.TimeSpan
+SkiaSharp.Extended.SKLottiePlayer.Render(SkiaSharp.SKCanvas! canvas, SkiaSharp.SKRect rect) -> void
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt:18]: Validated this second range as well: the names in the comment are identical, the net and netstandard Unshipped entries match, and CI-mode builds pass the PublicApiAnalyzers for both target frameworks without errors. These are the exact compiled API names; no correction is needed.

File: source/SkiaSharp.Extended.UI.Maui/Controls/Lottie/SKLottieImageSource.shared.cs [hunk 2]
```diff
@@ -52,12 +45,30 @@ public static object FromStream(Func<CancellationToken, Task<Stream?>> getter) =
...
+		while ((read = stream.Read(buffer, 0, buffer.Length)) > 0)
+		{
+			if (snapshot.Length + read > maximumSnapshotBytes)
+				throw new ArgumentException("The stream exceeds the 32 MiB Lottie source limit.", nameof(stream));
+			snapshot.Write(buffer, 0, read);
+		}
+
+		var bytes = snapshot.ToArray();
+		return FromStream(_ => Task.FromResult<Stream?>(new MemoryStream(bytes, writable: false)));
+	}
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended.UI.Maui/Controls/Lottie/SKLottieImageSource.shared.cs:78]: Fixed in `586373da69`: the bounded synchronous snapshot now retains the `MemoryStream` backing buffer via `TryGetBuffer` instead of duplicating it with `ToArray()`, and each load gets its own read-only stream. Documented that this legacy synchronous overload can block and recommends the asynchronous factory overload for slow streams. Added coverage proving replay works even after the caller disposes the original stream.

File: source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt [hunk 3]
```diff
@@ -1 +1,37 @@
 #nullable enable
+SkiaSharp.Extended.SKLottieAnimation
+SkiaSharp.Extended.SKLottieAnimation.Animation.get -> SkiaSharp.Skottie.Animation?
+SkiaSharp.Extended.SKLottieAnimation.IsLoaded.get -> bool
+SkiaSharp.Extended.SKLottieAnimation.SKLottieAnimation() -> void
+SkiaSharp.Extended.SKLottieAnimation.SKLottieAnimation(SkiaSharp.Skottie.Animation? animation) -> void
+SkiaSharp.Extended.SKLottiePlayer
+SkiaSharp.Extended.SKLottiePlayer.Animation.get -> SkiaSharp.Skottie.Animation?
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt:8]: Checked the actual entries against the compiled symbols: the alleged `SkiaSharp` vs `SkiaSharp` spellings in this comment are identical, and these entries match the generated analyzer format. `CI=true dotnet build source/SkiaSharp.Extended/SkiaSharp.Extended.csproj -c Release` validates both netstandard2.0 and net10.0 with zero analyzer errors; changing the names would break the baseline. No edit needed.

File: source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+			AnimationUpdated?.Invoke(this, EventArgs.Empty);
+			return;
+		}
+
+		if (deltaTime < TimeSpan.Zero)
+		{
+			SetPosition(Progress + TimeSpan.FromTicks(scaledTicks));
+			AnimationUpdated?.Invoke(this, EventArgs.Empty);
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs:125]: Fixed in `586373da69`. Negative elapsed time still intentionally acts as a signed seek without consuming repeat cycles; the target now clamps using boundary comparisons before adding ticks, so `TimeSpan.MinValue` cannot overflow and a negative speed correctly moves forward. Added tests for both extreme directions and negative-delta/negative-speed interaction.

File: source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt [hunk 5]
```diff
@@ -1 +1,37 @@
...
+SkiaSharp.Extended.SKLottiePlayer.Progress.get -> System.TimeSpan
+SkiaSharp.Extended.SKLottiePlayer.Render(SkiaSharp.SKCanvas! canvas, SkiaSharp.SKRect rect) -> void
+SkiaSharp.Extended.SKLottiePlayer.Repeat.get -> SkiaSharp.Extended.SKLottieRepeat
+SkiaSharp.Extended.SKLottiePlayer.Repeat.set -> void
+SkiaSharp.Extended.SKLottiePlayer.SKLottiePlayer() -> void
+SkiaSharp.Extended.SKLottiePlayer.Seek(System.TimeSpan position) -> void
+SkiaSharp.Extended.SKLottiePlayer.Update(System.TimeSpan deltaTime) -> void
+SkiaSharp.Extended.SKLottieRepeat
+SkiaSharp.Extended.SKLottieRepeat.Count.get -> int
+SkiaSharp.Extended.SKLottieRepeat.Equals(SkiaSharp.Extended.SKLottieRepeat other) -> bool
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/PublicAPI/netstandard/PublicAPI.Unshipped.txt:26]: Validated the remaining entries against the compiled API via `CI=true` Release core build for netstandard2.0 and net10.0. No PublicAPI analyzer errors; the spellings cited in the comment are identical. The API ledger is correct as written.

File: source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+	{
+		var scaledTicks = ticks * speed;
+		if (double.IsNaN(scaledTicks))
+			return 0;
+		if (scaledTicks >= long.MaxValue)
+			return long.MaxValue;
+		if (scaledTicks <= long.MinValue)
+			return long.MinValue;
+		return (long)scaledTicks;
+	}
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs:321]: Added focused regression coverage in `586373da69` for NaN/zero/positive and negative Infinity speed, `TimeSpan.MaxValue` with finite restart/reverse repeats, `TimeSpan.MinValue` negative deltas with both speed signs, and negative-speed finite restart/reverse completion. Also fixed the overflow found by those boundary cases. CI-mode core Lottie tests pass (20 tests) and MAUI Lottie tests pass (34 tests).

File: source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+	/// <summary>Fires after each <see cref="Seek"/> or <see cref="Update"/> call that changes playback state.</summary>
+	public event EventHandler? AnimationUpdated;
+
+	/// <summary>
+	/// Seeks the animation to an absolute position and raises <see cref="AnimationUpdated"/>.
+	/// </summary>
+	/// <remarks>
+	/// The position is clamped to [<see cref="TimeSpan.Zero"/>, <see cref="Duration"/>]. Seeking
+	/// never consumes repeats, changes direction, or raises <see cref="AnimationCompleted"/>. It
+	/// clears <see cref="IsComplete"/> so playback can continue from the requested frame.
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply source/SkiaSharp.Extended/Lottie/SKLottiePlayer.cs]: Agreed; updated the XML contract in `586373da69` to state that every `Seek` and processed `Update` notifies, even if the position stays the same, while updates without an animation or after completion are ignored. Added a test for no-op, completion, and ignored-update notifications.
